### PR TITLE
Request for Similar Embedding Works, with content query param

### PR DESCRIPTION
**Pull request by sshivaditya2019**

### CODE REVIEW

File: packages/engine/src/nodes/events/EventSearch.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  MagickWorkerInputs,
+  MagickWorkerOutputs,
+} from '../../types'
+import { InputControl } from '../../dataControls/InputControl'
+import { triggerSocket, stringSocket, eventSocket, arraySocket, anySocket } from '../../sockets'
+import { MagickComponent } from '../../magick-component'
+import { API_ROOT_URL } from '../../config'
+
+
+const info = 'Event Search is to find simlar Events '
```
Comment (lalalune) [inline packages/engine/src/nodes/events/EventSearch.ts:18]: similar

File: packages/plugin-discord/src/components/agent.component.tsx [hunk 2]
```diff
@@ -7,7 +7,7 @@ type PluginProps = {
   agentData: any
   props
 }
-import {API_ROOT_URL} from '@magickml/engine'
+import {API_ROOT_URL} from '../../../engine/src'
```
Comment (lalalune) [inline packages/plugin-discord/src/components/agent.component.tsx:10]: we shouldnt need this, might be an issue with our routes

**Comment by lalalune** [comment]
@sshivaditya looks good but we can probably consolidate this into the "event recall" node and only perform the search if the embedding field is not null or not null vector (array dim 1536 of all 0s)

File: packages/plugin-discord/src/components/agent.component.tsx (hunk 2 above)
Comment (sshivaditya2019) [reply packages/plugin-discord/src/components/agent.component.tsx:10]: Adding `@magickml/engine` causes circular dependency. The plugin.ts from engine package depends on Plugin_Discord which would now depend on `@magickml/engine`, if this is added, causing circular dependency

**Comment by sshivaditya2019** [comment]
@lalalune Merged with EventRecall, both are serviced using an after hook from Event Service.

File: packages/plugin-discord/src/components/agent.component.tsx (hunk 2 above)
Comment (lalalune) [reply packages/plugin-discord/src/components/agent.component.tsx:10]: Dang that's so annoying :P

**Pull request merged**